controllers/predictor_controller.go

```diff
@@ -239,6 +240,33 @@ func validatePredictor(predictor *api.Predictor) string {
 		return "Only one of spec.schemaPath and spec.storage.schemaPath can be specified"
 	}
 
+	if storage.StorageSpec.Parameters != nil {
+		parameters := *storage.StorageSpec.Parameters
+		if storageType, ok := parameters["type"]; ok && storageType == StoragePVCType {
+			s := &corev1.Secret{}
+			if err := pr.Client.Get(context.TODO(), types.NamespacedName{
+				Name:      modelmesh.StorageSecretName,
+				Namespace: predictor.Namespace,
+			}, s); err != nil {
+				return "Could not get the storage secret"
+			}
+			var storageConfig map[string]string
+			found := false
+			for _, storageData := range s.Data {
+				if err := json.Unmarshal(storageData, &storageConfig); err != nil {
+					return "Could not parse storage configuration json"
+				}
+				if storageConfig["type"] == StoragePVCType && storageConfig["name"] == parameters["name"] {
```

Comment:
We should check for the existence of the "name" key in both cases... with a wrong configuration this could panic.